src/components/slider/slider.ts

```diff
@@ -0,0 +1,251 @@
+  get value() {
+    return this._value;
+  }
+
+  set value(v: number) {
+    this._value = Number(v);
+    this._isInitialized = true;
+    this.updatePercentFromValue();
+  }
+
+  constructor(elementRef: ElementRef) {
+    this._renderer = new SliderRenderer(elementRef);
+  }
+
+  /**
+   * Once the slider has rendered, grab the dimensions and update the position of the thumb and
+   * fill track.
+   * TODO: internal
+   */
+  ngAfterContentInit() {
+    this._sliderDimensions = this._renderer.getSliderDimensions();
+    this._renderer.updateThumbAndFillPosition(this._percent, this._sliderDimensions.width);
+  }
+
+  /** TODO: internal */
+  onClick(event: MouseEvent) {
+    if (this.disabled) {
+      return;
+    }
+
+    this.isActive = true;
+    this.isDragging = false;
+    this._renderer.addFocus();
+    this.updateValueFromPosition(event.clientX);
+  }
+
+  /** TODO: internal */
+  onDrag(event: HammerInput) {
+    if (this.disabled) {
+      return;
+    }
+    // Prevent the drag from selecting anything else.
+    event.preventDefault();
+    this.updateValueFromPosition(event.center.x);
+  }
+
+  /** TODO: internal */
+  onDragStart(event: HammerInput) {
+    if (this.disabled) {
+      return;
+    }
+
+    event.preventDefault();
+    this.isDragging = true;
+    this.isActive = true;
+    this._renderer.addFocus();
+    this.updateValueFromPosition(event.center.x);
+  }
+
+  /** TODO: internal */
+  onDragEnd() {
+    this.isDragging = false;
+  }
+
+  /** TODO: internal */
+  onResize() {
+    this.isDragging = true;
+    this._sliderDimensions = this._renderer.getSliderDimensions();
+    // Skip updating the value and position as there is no new placement.
+    this._renderer.updateThumbAndFillPosition(this._percent, this._sliderDimensions.width);
+  }
+
+  /** TODO: internal */
+  onBlur() {
+    this.isActive = false;
+  }
+
+  /**
+   * When the value changes without a physical position, the percentage needs to be recalculated
+   * independent of the physical location.
+   */
+  updatePercentFromValue() {
+    this._percent = (this.value - this.min) / (this.max - this.min);
+  }
+
+  /**
+   * Calculate the new value from the new physical location.
+   */
+  updateValueFromPosition(pos: number) {
+    let offset = this._sliderDimensions.left;
+    let size = this._sliderDimensions.width;
+    this._percent = this.clamp((pos - offset) / size);
+    this.value = this.min + (this._percent * (this.max - this.min));
+
+    this._renderer.updateThumbAndFillPosition(this._percent, this._sliderDimensions.width);
+  }
+
+  /**
+   * Return a number between two numbers.
+   */
+  clamp(value: number, min = 0, max = 1) {
+    return Math.max(min, Math.min(value, max));
+  }
+}
+
+/**
+ * Renderer class in order to keep all dom manipulation in one place and outside of the main class.
+ */
+export class SliderRenderer {
+  private _sliderElement: HTMLElement;
+
+  constructor(private _elementRef: ElementRef) {
```

Comment:
Is it necessary to save `_elementRef` on the class? Looks like you are only using it for `this._sliderElement`.